src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py

```diff
@@ -41,10 +40,10 @@ def __init__(self, input_h, input_w, batch_size, output_size, init_weights, conf
         self.is_train = is_train  # Set flag for training or val
 
         # Create placeholders for image and labels
-        self.images = _tf.placeholder(_tf.float32, [self.batch_size, input_h,
+        self.images = _tf.placeholder(_tf.float32, [None, input_h,
                                                               input_w, 3], name='images')
         self.labels = _tf.placeholder(_tf.float32,
-                                [self.batch_size, self.grid_shape[0], self.grid_shape[1],
+                                [None, self.grid_shape[0], self.grid_shape[1],
```

Comment:
Just curious, if you fix batch size, why you need to default it to `None`?

Reply:
So here I changed it to be a dynamic batch size, by declaring the shape dimension to be None.

Reply:
Oh well, I think you can still pass batch_size but in your backend, you use actual size. This means the inference for the right size is mandatory and you don't need to explicitly enable it.